AdyenComponents/BACS Direct Debit/Trackers/BACSDirectDebitComponentTracker.swift

```diff
@@ -8,35 +8,37 @@
 import Foundation
 
 internal protocol BACSDirectDebitComponentTrackerProtocol: AnyObject {
-    func sendTelemetryEvent()
+    func sendInitialAnalytics()
 }
 
 internal class BACSDirectDebitComponentTracker: BACSDirectDebitComponentTrackerProtocol {
 
     // MARK: - Properties
 
     private let paymentMethod: BACSDirectDebitPaymentMethod
-    private let apiContext: APIContext
-    private let telemetryTracker: TelemetryTrackerProtocol
+    private let context: AdyenContext
     private let isDropIn: Bool
 
     // MARK: - Initializers
 
     internal init(paymentMethod: BACSDirectDebitPaymentMethod,
-                  apiContext: APIContext,
-                  telemetryTracker: TelemetryTrackerProtocol,
+                  context: AdyenContext,
                   isDropIn: Bool) {
         self.paymentMethod = paymentMethod
-        self.apiContext = apiContext
-        self.telemetryTracker = telemetryTracker
+        self.context = context
         self.isDropIn = isDropIn
     }
 
     // MARK: - BACSDirectDebitComponentTrackerProtocol
 
-    internal func sendTelemetryEvent() {
-        let flavor: TelemetryFlavor = isDropIn ? .dropInComponent : .components(type: paymentMethod.type)
-        telemetryTracker.sendTelemetryEvent(flavor: flavor)
+    internal func sendInitialAnalytics() {
+        // initial call is not needed again if inside dropIn
+        guard !isDropIn else { return }
+        let flavor: AnalyticsFlavor = .components(type: paymentMethod.type)
+        let amount = context.payment?.amount
+        let additionalFields = AdditionalAnalyticsFields(amount: amount, sessionId: AnalyticsForSession.sessionId)
+        context.analyticsProvider?.sendInitialAnalytics(with: flavor,
+                                                         additionalFields: additionalFields)
```

Comment:
Can we make `BACSDirectDebitComponentTracker ` implement `TrackableComponent` ?

Reply:
But this class is not a component :)  BACS is written a little differently like a VIPER structure.

Reply:
But only responsibility of `BACSDirectDebitComponentTracker ` is to send analytics.
If `BACSDirectDebitComponent` will implements `TrackableComponent` then we can pass it to `BACSInputPresenter` as `TrackableComponent`. 
This way we keep `VIPER structure` and reuse existing code for sending analytics.